tests/_test_mscolab/test_server.py

```diff
@@ -130,14 +132,45 @@ def test_get_user(self):
     def test_delete_user(self):
         assert add_user(self.userdata[0], self.userdata[1], self.userdata[2])
         with self.app.test_client() as test_client:
+            # Case 1 : The user has no profile image set
             token = self._get_token(test_client, self.userdata)
             response = test_client.post('/delete_own_account', data={"token": token})
             assert response.status_code == 200
-            data = json.loads(response.data.decode('utf-8'))
-            assert data["success"] is True
-            response = test_client.post('/delete_own_account', data={"token": "dsdsds"})
+            assert response.get_json()["success"] is True
+            # ToDo: Check if user token was cleared after deleting account as assert returns True instead of False
+            # assert verify_user_token(config_loader(dataset="mscolab_server_url"), token) is False
+
+            # Case 2 : The user has a custom profile image set
+            assert add_user(self.userdata[0], self.userdata[1], self.userdata[2])
+            token = self._get_token(test_client, self.userdata)
+            response = self._upload_profile_image(test_client, token, self.userdata[0])
+            assert response.status_code == 200  # this will ensure image was uploaded
+            response = test_client.post('/delete_own_account', data={"token": token})
             assert response.status_code == 200
-            assert response.data.decode('utf-8') == "False"
+            assert response.get_json()["success"] is True
+            # ToDo: Check if user token was cleared after deleting account as assert returns True instead of False
+            # assert verify_user_token(config_loader(dataset="mscolab_server_url"), token) is False
+
+    # ToDo: Add a test for an oversized image/file ( > MAX_UPLOAD_SIZE) for chat attachments and profile image.
+    # Currently, flask is unable to raise exception for an oversized file.
+
+    def test_unauthorized_profile_image_upload(self):
+        other_user_data = 'other@ex.com', 'other', 'other'
+        assert add_user(self.userdata[0], self.userdata[1], self.userdata[2])
+        assert add_user(other_user_data[0], other_user_data[1], other_user_data[2])
+        with self.app.test_client() as test_client:
+            # Case 1: Unauthenticated upload attempt
+            user = get_user(self.userdata[0])
+            assert user.profile_image_path is None
+            self._upload_profile_image(test_client, token="random-string", email=self.userdata[0])
```

Comment:
This random-string does not exist and does not depend on the SECRET_KEY, so it can't get further.

Reply:
isn't that what we want?

Reply:
this now tests verify_user used, so yes we can keep it. This was requested.